### DISCUSSION TITLE
python can library error on blf writer

### DISCUSSION THREAD

**Discussion by AntonioMembrino** [extremecoders-re/pyinstxtractor#72] [General]
hi all, I would log messages of a virtua can on a blf file. I'm using python can library. there is an error on the function write of BlfWriter.
i report here  the script 
` import can

def create_blf_logging(log_file_path, can_interface):
    # Set up the virtual CAN interface
    bus = can.interface.Bus(channel=can_interface, bustype='socketcan')

    # Create the BLF log file
    blf_writer = can.io.BLFWriter(log_file_path)
    blf_logger= can.io.logger
    try:
        # Start receiving and logging CAN messages
        while True:
            message = bus.recv(timeout=1.0)  # Set the timeout value based on your application needs
            if message is not None:
                # Print the raw CAN message for verification (optional)
                print(f"Arbitration ID: {message.arbitration_id}, Data: {message.data.hex()}")

                # Write the raw CAN message to the BLF log file
                blf_writer.write(message)

    except KeyboardInterrupt:
        print("Logging stopped by the user.")
    finally:
        # Close the log file and clean up resources
        blf_writer.stop()
        bus.shutdown()

if __name__ == "__main__":
    log_file_path = "log_file.blf"
    can_interface = "can1"  # Use the virtual CAN interface (vcan0)
    create_blf_logging(log_file_path, can_interface)`

**the error is on the write method** why   : 'BLFWriter' object has no attribute 'write'?
how can I write on the file?

**Comment by extremecoders-re** [comment]
This does not seem to be related to this project - pyinstaller extractor.
Perhaps you are looking for https://github.com/hardbyte/python-can
